app/javascript/pages/country/widget/widgets/widget-tree-gain/widget-tree-gain-actions.js

```diff
@@ -14,15 +14,24 @@ const getTreeGain = createThunkAction(
     if (!state().widgetTreeGain.loading) {
       dispatch(setTreeGainLoading({ loading: true, error: false }));
       axios
-        .all([getGain({ ...params }), getExtent({ ...params })])
+        .all([
+          getGain({ ...params }),
+          getExtent({ ...params }),
+          getGainRanking({ ...params })
+        ])
         .then(
-          axios.spread((gainResponse, extentResponse) => {
+          axios.spread((gainResponse, extentResponse, gainRankingResponse) => {
```

Comment:
Im not sure an extra query is needed here. As we have to create a ranking list, surely a single fetch for all countries data is needed and then the selectors do the work?